src/algorithms/meta/SubDivideFunctors.h

```diff
@@ -90,23 +105,33 @@ template <auto... MemberFunctionPtrs>
 class ValueSplit {
 public:
 
-    ValueSplit(std::vector<std::vector<int>> ids) : m_ids(ids) {};
+    ValueSplit( std::vector<std::vector<int>> ids, bool matching = true)
```

Comment:
Also, optionally, I've wondered about a `TrueFalseSplit` functor a number of times. That's really just a `ValueSplit` (which is how this gets implemented in plugins), but it could have the benefit of accepting a function that returns a bool, which is semantically cleaner than relying on bool to integer conversions.

Reply:
You would have to move away from the simple std::find implementation for anything other than ==. I'll have another think, as changing the functors already there is making them less clear. It should be possible to pass an array of boolean operators along with values and member functions to a functor.

After the vertexing discussion, access to min/max might be nice too.

Reply:
BooleanSplit (name can still be changed) should be able to perform the jobs of RangeSplit and ValueSplit where an array of functions which return a bool from two inputs is passed alongside the array of values to compare. The bools are all reduced by **and** rather than trying to mix in some more complexity.

What is missing the the ability to compare types other than float, I'm not sure how to approach for instance asking for e.g. ReconstructedParticles with PDG "equal_to" and a goodnessOfPID "greater_than". Maybe it's not important, could be done in a few steps, or this sort of thing is not what we'll want to do in eicrecon.